scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -57,11 +58,11 @@ def _highly_variable_genes_seurat_v3(
         )
 
     X = adata.layers[layer] if layer is not None else adata.X
-    if check_nonnegative_integers(X) is False:
-        raise ValueError(
-            "`pp.highly_variable_genes` with `flavor='seurat_v3'` expects "
-            "raw count data."
-        )
+    if check_values:
+        if check_nonnegative_integers(X) is False:
+            logg.warning(
+                "`flavor='seurat_v3'` expects raw count data, but non-integers were found."
+            )
```

Comment:
Can this be an actual [`warning`](https://docs.python.org/3/library/warnings.html)?

Reply:
Logging frameworks should have a way of recording warnings on their own. In general, we should not be using `logg.warning`

Reply:
```suggestion
           warn(
                "`flavor='seurat_v3'` expects raw count data, but non-integers were found.",
                UserWarning
            )
```

Reply:
done, and added test